fix: increment error caching counter when trace not sampled (#10631)
grafana/mimir: CHANGELOG.md

* [BUGFIX] Distributor: Use a boolean to track changes while merging the ReplicaDesc components, rather than comparing the objects directly. #10185
* [BUGFIX] Querier: fix timeout responding to query-frontend when response size is very close to `-querier.frontend-client.grpc-max-send-msg-size`. #10154
* [BUGFIX] Query-frontend and querier: show warning/info annotations in some cases where they were missing (if a lazy querier was used). #10277

> * [BUGFIX] Query-frontend: Fix an issue where transient errors could be inadvertently cached. #10537 #10631

Reviewer: ```suggestion
* [BUGFIX] Query-frontend: Fix an issue where transient errors are inadvertently cached. #10537 #10631
```
